### PR TITLE
Supports eip 3770 address format, better network support

### LINKED ISSUE #338: Correct Native Token Symbol on all networks
Based on Feedback [here](https://github.com/gnosis/safe-apps-list/issues/92#issuecomment-1011916061)
> The summary for native tokens on Xdai, Polygon & BSC shows them as ETH and I think it would be nice to provide feedback per native token

We do already have a network map with native token symbols in the code. I believe this was working in the past, but we must have accidentally removed this...

### PR DESCRIPTION
**Pull request by schmanu**
Other changes:
* network specifics moved into one data structure
* corrects token symbols for all supported networks
* drain safe generation skips 0 balances of native tokens
* removes faulty mergify config
* makes transfers backward compatible. If no token_type is given we assume erc20 / native.

closes #338, #322

### CODE REVIEW

File: src/networks.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      baseAPI: "https://safe-transaction.gnosis.io/api/v1",
+    },
+  ],
+  [
+    4,
+    {
+      chainID: 4,
+      name: "Rinkeby",
+      shortName: "rin",
+      currencySymbol: "RIN",
```
Comment (bh2smith) [inline src/networks.ts:26]: I believe the native token on the test nets are all still ETH

File: src/networks.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      currencySymbol: "BNB",
+      baseAPI: "https://safe-transaction.bsc.gnosis.io/api/v1",
+    },
+  ],
+  [
+    100,
+    {
+      chainID: 100,
+      name: "Gnosis Chain (formerly xDai)",
+      shortName: "xdai", // gno this is the offical shortname. wierdly gnosis Safe still uses xdai
```
Comment (bh2smith) [inline src/networks.ts:65]: Not sure I understand the comment here. Short name is that the short name for the chain or the currency?

**Review by bh2smith — APPROVED** [review]
Looks good except the currency names on rinkeby, kovan and other eth test nets.

File: src/networks.ts (hunk 1 above)
Comment (schmanu) [reply src/networks.ts:26]: I was very confused by what standards are.

There is https://eips.ethereum.org/EIPS/eip-3770 this which is implemented by the gnosis safe.

There it says to checkout the shortnames the full list is defined in https://github.com/ethereum-lists/chains

There again it says
>There are also aggregated json files with all chains automatically assembled:
    - https://chainid.network/chains.json

In this json are definitions for each chainID. Including the currencySymbols. Thats where I found them. So I think this should be the correct currency. But I was confused at first too.

Comment (bh2smith) [reply src/networks.ts:26]: It's fine. I call it rinkeby eth anyway so I guess it starts with RIN anyway. Must be metamask who is out of date.

File: src/networks.ts (hunk 2 above)
Comment (schmanu) [reply src/networks.ts:65]: It's the shortname for the chain. It is used for the eip3770 address format i.e. `xdai:0x8e6c423585eC174306caF372132663c00Df877a3`.
According to the standard it should already be `gno:0x8e6c423585eC174306caF372132663c00Df877a3`. 

But I'll keep it like this for the moment.

Comment (bh2smith) [reply src/networks.ts:65]: Works for me!

**Pull request merged**